Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -45,46 +45,72 @@ UPSWorldSubsystem& UPSWorldSubsystem::Get(const UObject& WorldContextObject)
 	return *ThisSubsystem;
 }
 
-// Returns current row of progression system
-const FPSRowData& UPSWorldSubsystem::GetCurrentRow() const
-{
-	const UPSSaveGameData* SaveGameInstance = GetCurrentSaveGameData();
-	checkf(SaveGameInstance, TEXT("ERROR: 'SaveGameInstanceInternal' is null"));
-
-	return SaveGameInstance->GetCurrentRow();
-}
-
 // Set current row of progression system by tag
 void UPSWorldSubsystem::SetCurrentRowByTag(FPlayerTag NewRowPlayerTag)
 {
-	// check if current row is similar
-	if (GetCurrentRow().Character == NewRowPlayerTag)
+	for (const auto& KeyValue : ProgressionSettingsDataInternal)
 	{
-		return;
-	}
+		const FPSRowData& RowData = KeyValue.Value;
 
-	UPSSaveGameData* SaveGameInstance = GetCurrentSaveGameData();
-	checkf(SaveGameInstance, TEXT("ERROR: 'SaveGameInstanceInternal' is null"));
-
-	SaveGameInstance->SetRowByTag(NewRowPlayerTag);
-	OnCurrentRowDataChanged.Broadcast(NewRowPlayerTag);
+		if (RowData.Character == NewRowPlayerTag)
+		{
+			CurrentRowNameInternal = KeyValue.Key;
+			OnCurrentRowDataChanged.Broadcast(NewRowPlayerTag);
+			return; // Exit immediately after finding the match
+		}
+	}
 }
 
-// Returns previous row of progression system
-const FPSRowData& UPSWorldSubsystem::GetPreviousRow() const
+// Returns previous row by current name
+FPSRowData& UPSWorldSubsystem::GetPreviousRow()
 {
-	const UPSSaveGameData* SaveGameInstance = GetCurrentSaveGameData();
-	checkf(SaveGameInstance, TEXT("ERROR: 'SaveGameInstanceInternal' is null"));
-
-	return SaveGameInstance->GetPreviousRow();
+	return GetPreviousRowByCurrentName();
 }
 
 // Returns the data asset that contains all the assets of Progression System game feature
 const UPSDataAsset* UPSWorldSubsystem::GetPSDataAsset() const
 {
+	ensureMsgf(PSDataAssetInternal, TEXT("ASSERT: [%i] %s:\n'PSDataAssetInternal' is empty!"), __LINE__, *FString(__FUNCTION__));
 	return PSDataAssetInternal.LoadSynchronous();
 }
 
+//  Returns a current save to disk row name
+const FName UPSWorldSubsystem::GetFirstSaveToDiskRowData() const
+{
+	ensureMsgf(SaveGameDataInternal, TEXT("ASSERT: [%i] %s:\n'SaveGameDataInternal' is empty!"), __LINE__, *FString(__FUNCTION__));
+	return SaveGameDataInternal->GetSavedProgressionRowByIndex(0);
+}
+
+//  Returns a current save to disk row by name
+FPSSaveToDiskData& UPSWorldSubsystem::GetCurrentSaveToDiskRowByName()
+{
+	ensureMsgf(SaveGameDataInternal, TEXT("ASSERT: [%i] %s:\n'SaveGameDataInternal' is empty!"), __LINE__, *FString(__FUNCTION__));
```

Comment:
`ensureMsgf` is recommended to use only with `if` statements, as it doesn't handle crashes. If `SaveGameDataInternal` is null, your code will crash on a nullptr. Use `ensureMsgf` inside an `if` statement before accessing the pointer, or use `checkf`, which will cause Unreal Engine to crash the game before the nullptr does, ensuring the crash is logged.

Since it's Getter function:
- I recommend just do `if` silently, but handle return result in caller functions with ensureMsgf
- Do the same for other Get functions above

### Expected result:
```
// Here in this Getter:
    if (SaveGameDataInternal)
    {
        return SaveGameDataInternal->GetSavedProgressionRowByIndex(0);
    }

    return NAME_None
```

```cpp
// Somewhere in caller functions:
const FName SomeNameFromSave = SaveInData.GetSomeName();
if (ensureMsgf(!SomeNameFromSave.IsNone(), TEXT("ASSERT: [%i] %s:\n'SaveInData.GetSomeName()' is None!"), __LINE__, *FString(__FUNCTION__)))
{
    // Safe to use SomeNameFromSave
    // Further logic
}
```

Reply:
done:  https://github.com/h4rdmol/ProgressionSystem/pull/18/commits/3f35b83bc75bb5b22dd3d7c846415f75ec2a4e6c